File: web/src/types/serviceAlert.ts
export interface InformedEntity {
  agency_id?: string
  route_id?: string
  route_type?: number
  stop_id?: string
  activities?: string[]
}

export interface ServiceAlert {
  alertId: string
  agency: 'sound_transit' | 'king_county_metro'
  effect?: string
  cause?: string
  headerText?: string
  descriptionText?: string
  severityLevel?: string
  activePeriodStart?: number
  activePeriodEnd?: number
  informedEntities?: InformedEntity[]
  url?: string
  lastSeen: string
}

File: api/internal/models/serviceAlert.go
package models

type ServiceAlert struct {
	AlertID           string      `json:"alertId"`
	Agency            string      `json:"agency"`
	Effect            string      `json:"effect,omitempty"`
	Cause             string      `json:"cause,omitempty"`
	HeaderText        string      `json:"headerText,omitempty"`
	DescriptionText   string      `json:"descriptionText,omitempty"`
	SeverityLevel     string      `json:"severityLevel,omitempty"`
	ActivePeriodStart *int64      `json:"activePeriodStart,omitempty"`
	ActivePeriodEnd   *int64      `json:"activePeriodEnd,omitempty"`
	InformedEntities  interface{} `json:"informedEntities"`
	URL               string      `json:"url,omitempty"`
	LastSeen          string      `json:"lastSeen"`
}
